tests/bsim/bluetooth/mesh/src/test_cdp1.c

```diff
@@ -0,0 +1,517 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Composition Data Page 1 (CDP1) test
+ */
+
+#include "mesh_test.h"
+
+#include <stddef.h>
+#include <string.h>
+#include <mesh/access.h>
+#include <mesh/net.h>
+
+#include <zephyr/logging/log.h>
+#define LOG_MODULE_NAME test_cdp1
+LOG_MODULE_REGISTER(LOG_MODULE_NAME, LOG_LEVEL_INF);
+
+#define NODE_ADDR 0x00a1
+#define WAIT_TIME 60 /* seconds */
+
+#define TEST_MODEL_ID_1 0x2a2a
+#define TEST_MODEL_ID_2 0x2b2b
+#define TEST_MODEL_ID_3 0x2c2c
+#define TEST_MODEL_ID_4 0x2d2d
+#define TEST_MODEL_ID_5 0x2e2e
+#define TEST_MODEL_ID_6 0x2f2f
+#define TEST_VND_MODEL_ID_1 0x3a3a
+
+static int model1_init(struct bt_mesh_model *model);
+static int model2_init(struct bt_mesh_model *model);
+static int model3_init(struct bt_mesh_model *model);
+static int model4_init(struct bt_mesh_model *model);
+static int model5_init(struct bt_mesh_model *model);
+static int model6_init(struct bt_mesh_model *model);
+static int model_vnd1_init(struct bt_mesh_model *model);
+
+static uint8_t app_key[16] = { 0xaa };
+static uint8_t net_key[16] = { 0xcc };
+
+static const struct bt_mesh_test_cfg node1_cfg = {
+	.addr = NODE_ADDR,
+	.dev_key = {0xaa},
+};
+
+static const struct bt_mesh_model_cb test_model1_cb = {
```

Comment:
A lot of copy-paste. Could you create a parametrized macro to create all necessary entities for one model?
Smth like:
```
#define MODEL_EXT(test_id_number) \
bt_mesh_model_extend(bt_mesh_model_find(&elems[0], test_id_number), model)

#define MODEL_NO_EXT  0

#define TEST_MODEL_DECLARE(number, ...) \
static const struct bt_mesh_model_cb test_model##number##_cb = { \
	.init = model##number##_init,    \
}; \
static const struct bt_mesh_model_op model_op##number##[] = { \
	BT_MESH_MODEL_OP_END, \
}; \
static int model##number##_init(struct bt_mesh_model *model) \
{ \
        ASSERT_OK(__VA_ARGS__); \
	return 0; \
}
```

Reply:
Made macros for declaring models. Thanks for the suggestion!

Reply:
I'm pretty sure it is possible to put functions under macro to have less copy-paste code. However, this is much better than nothing.